Rutorrent GetMulitpleTorrents.user.js

```diff
@@ -27,17 +27,36 @@ if (nav_bar) {
 }
 
 function getTorrents() {
-  var selected = document.getElementsByClassName("selected");
-  var dl_link = 'plugins/source/action.php?hash=';
+  const selected = document.getElementsByClassName("selected");
+  let dl_link = 'plugins/source/action.php';
 
   if ('/' != window.location.toString().slice(-1)) {
     dl_link = '/' + dl_link;
   }
 
-  for (var i = 0; i < (selected.length - 1); i++) {
-    var hash = selected[i].id;
+  function downloadTorrent(hash) {
+    const form = document.createElement("form");
```

Comment:
Maybe this works too? Would be more direct

```
  let xhttp = new XMLHttpRequest();
  xhttp.open("POST", dl_link , true);
  // xhttp.setRequestHeader("Content-type", "application/x-www-form-urlencoded");   // maybe?
  xhttp.send("fame=" + hash);
```

Reply:
Going with AJAX was also my first intuition. However, unfortunately, when using AJAX the response does not trigger a download action by itself.
To make it work ultimately I would still need to create a link and trigger a fake click event or resort to other ugly workarounds with iframes or window.open.

Have a look at: https://stackoverflow.com/questions/16086162/handle-file-download-from-ajax-post

Ultimately, while the form submission tech feels hacky, I couldn't come with a non-hacky solution.